src/v/coproc/tests/utils/event_publisher.cc

```diff
@@ -140,13 +119,19 @@ ss::future<std::vector<kafka::produce_response::partition>>
 event_publisher::publish_events(model::record_batch_reader reader) {
     return std::move(reader)
       .for_each_ref(
-        coproc::reference_window_consumer(
-          coproc::wasm::batch_verifier(),
-          coproc::wasm::publisher(_client, model::coprocessor_internal_tp)),
-        model::no_timeout)
-      .then([](auto tuple) {
-          vassert(std::get<0>(tuple), "crc checks failed");
-          return std::move(std::get<1>(tuple));
+        storage::internal::decompress_batch_consumer(), model::no_timeout)
+      .then([this](model::record_batch_reader rbr) {
+          return std::move(rbr)
+            .for_each_ref(
+              coproc::reference_window_consumer(
+                coproc::wasm::batch_verifier(),
+                coproc::wasm::publisher(
+                  _client, model::coprocessor_internal_tp)),
+              model::no_timeout)
+            .then([](auto tuple) {
+                vassert(std::get<0>(tuple), "crc checks failed");
+                return std::move(std::get<1>(tuple));
```

Comment:
this screams at the reader for a named struct rather than a tuple. Tuples are a poor man's structs.

Reply:
This is due to some template hackery. You can pass any list of arbitrary types as the input here: https://github.com/vectorizedio/redpanda/blob/dev/src/v/coproc/reference_window_consumer.hpp#L72
so the output may this odd aggregate type depending on the input